flytekit/extras/tasks/shell.py

```diff
@@ -210,8 +217,22 @@ def execute(self, **kwargs) -> typing.Any:
             print(gen_script)
             print("\n==============================================\n")
 
+        if self._env is not None:
+            for k, v in self._env.items():
+                os.environ[k] = str(v)
+
+        working_dir = flytekit.current_context().working_directory
+        tmp_script_path = os.path.join(working_dir, "tmp_script.sh")
+        with open(tmp_script_path, 'w') as fp:
+            fp.write(gen_script)
+        os.chmod(tmp_script_path, stat.S_IRWXU)
+        if self._script_args:
+            full_script = tmp_script_path + " " + self._script_args
+        else:
+            full_script = tmp_script_path
+
         try:
-            subprocess.check_call(gen_script, shell=True)
+            subprocess.run(full_script, shell=True, check=True)
```

Comment:
Oof. bad job flight with no check=True. Thanks for fixing that!

Reply:
I think `check_call()` is an older method that has `check=True` by default. I beleive all the various subprocess methods were consolidated and `run()` is now the recommended method with all the other methods collapsed into params